src/olympia/addons/tests/test_views.py

```diff
@@ -377,6 +377,26 @@ def _set_tested_url(self, param):
         self.url = reverse_ns(
             'addon-detail', api_version='v5', kwargs={'pk': param})
 
+    def test_queries(self):
+        with self.assertNumQueries(16):
+            # 16 queries
+            # - 2 savepoints because of tests
+            # - 1 for the add-on
+            # - 1 for its translations
+            # - 1 for its categories
+            # - 1 for its current_version
+            # - 1 for translations of that version
+            # - 1 for applications versions of that version
+            # - 1 for files of that version
+            # - 1 for authors
+            # - 1 for previews
+            # - 1 for license
+            # - 1 for translations of the license
+            # - 1 for discovery item (is_recommended)
+            # - 1 for featured collection presence (is_featured)
+            # - 1 for tags
+            self._test_url(lang='en-US')
```

Comment:
Is this actually a valid test-case for the actual issue? Unless I'm mistaken we were hitting "duplicated" queries that multiplied by the amount of add-ons shown right? We aren't creating any additional add-ons here to verify that that's no longer the case.

Reply:
Also, can we easily `select_related` any of the other relations without hurting this or other APIs?

Reply:
That one specifically is not for the valid issue, it's the add-on detail API. I made it to make sure I didn't regress anything while fixing the versions API.

There are very few single relations in there, so few opportunities for a `select_related()`. I think the only case where that'd work is discovery item - there is also the current version, but we'd skip the Version `transformer` so I don't know what that would do. In any case that'd be a different issue.